pkg/controller/registry/resolver/operators.go

```diff
@@ -225,16 +233,25 @@ type OperatorSurface interface {
 	SourceInfo() *OperatorSourceInfo
 	Bundle() *api.Bundle
 	Inline() bool
+	VersionDependencies() []VersionDependency
 }
 
 type Operator struct {
-	name         string
-	replaces     string
-	providedAPIs APISet
-	requiredAPIs APISet
-	version      *semver.Version
-	bundle       *api.Bundle
-	sourceInfo   *OperatorSourceInfo
+	name                string
+	replaces            string
+	providedAPIs        APISet
+	requiredAPIs        APISet
+	version             *semver.Version
+	bundle              *api.Bundle
+	sourceInfo          *OperatorSourceInfo
+	versionDependencies []VersionDependency
+}
+
+type VersionDependency struct {
+	Package string
+	Version semver.Version
```

Comment:
Are we supposed to handle the `semver.Range` as well given users can specify version of a package as a semver range.